artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/impl/PagingStoreImpl.java

```diff
@@ -290,7 +296,11 @@ public void applySetting(final AddressSettings addressSettings) {
       }
 
       if (pageLimitBytes != null && pageSize > 0) {
+         Long originalEstimatedMaxPages = this.estimatedMaxPages;
          estimatedMaxPages = pageLimitBytes / pageSize;
+         if (!Objects.equals(estimatedMaxPages, originalEstimatedMaxPages)) {
+            checkNumberOfPages();
+         }
```

Comment:
I would move this below the log statement, otherwise it has possibly gone away and done stuff based on the change, before logging what it was.

I also dont think it should ever be doing this for the constructor case, as it hasnt finished creating the object yet and is pre-empting start() doing this later. The earlier change for message count kind of gets away with it due to the null check on cursorProvider (which is actually final and so only could be null during the constructor), but its not clear this one does. It might be time to pass a boolean to indicate whether this shared method is being called from the constructor or later, so it simply cant do these things at construction (allowing removal of the null check)

I think this bit, and the pageLimitMessages related check earlier, should perhaps just set booleans to indicate if they changed, which you then action at the end after both have been updated, by calling checkNumberOfPages() and cursorProvider.checkClearPageLimit() respectively (again, if not acting for the constructor)